### ISSUE TITLE
Support multiple deserialize renames for the same field

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/serde#595]
``` rust
#[derive(Serialize, Deserialize)]
struct S {
    #[serde(rename(deserialize = "old_name", deserialize = "new_name"))]
    new_name: u8
}
```

Asked by @SuperFluffy in #rust.

**Comment by dtolnay** [comment]
@SuperFluffy for now the workaround would be to implement Deserialize something like this:

```rust
impl Deserialize for S {
    fn deserialize<D>(deserializer: &mut D) -> Result<Self, D::Error>
        where D: Deserializer
    {
        #[derive(Deserialize)]
        struct Helper {
            old_name: Option<u8>,
            new_name: Option<u8>,
        }

        let helper: Helper = try!(Deserialize::deserialize(deserializer));

        let s = /* do the conversion, maybe fail if both are set */;
        Ok(s)
    }
}
```

**Comment by scull7** [comment]
I believe this would be useful with type of structure also: https://github.com/serde-rs/serde/issues/970#issuecomment-438153519

**Comment by dtolnay** [comment]
Implemented in #1458.
